src/tools/utils.py

```diff
@@ -20,6 +21,33 @@
 NON_TERMINAL_STATES = {"Created", "Queued", "Running"}
 
 
+@beartype
+def query_run_statuses(job_ids: list[str]) -> dict:
+    """get statuses for multiple jobs in parallel
+
+    Args:
+        job_ids (list[str]): list of job IDs
+
```

Comment:
this should theoretically take the same amount of time no matter how many requests